### PR TITLE
Refactor for new standalone_resource layout

### PR DESCRIPTION
**Pull request by SteveALee**
Note that `standalone_resource` layout is identical to that in `supplemental-quidance` and both will soon be moved the `wai-website-theme`

### CODE REVIEW

**Review by WilcoFiers — CHANGES_REQUESTED** [review]
I showed the ACT TF the updated designs during today's call. I wanted to make sure you receive the feedback on it. 

- The "About Rules" is missing from the navigation
- The style on the active page is missing
- We're not clear on the "For developers of evaluation tools and test methodologies" phrase. Is the intent for each page to have the target audience identified in the header? If so we want to discuss this in more details. We briefly talked about "Accessibility testers and test writers", but haven't settled.
- Test in the header isn't correctly aligned.
- The empty right column doesn't seem like good use of space. At least some of us preferred the sidebar having the table of contents there.
- The top part of the page seems repetitive. This isn't new, but having "Test Rules" repeated 3 times; the header, then the nav, then the h1, that seemed to some of us unnecessarily repetitive. 
- The table of contents on the rules pages is missing the "Test Rules describe ways to test conformance to WCAG success criteria. They are for evaluation." sentence that was there before. Was that intentional?

**Comment by SteveALee** [comment]
Thanks for the feedback. [Adding @shawna-slh))]

> The "About Rules" is missing from the navigation

It was moved above to the context links. Maybe one to revisit.


> Test in the header isn't correctly aligned.

That's a strange bug as copied from suppl guidance.

> The table of contents on the rules pages is missing the "Test Rules describe ways to test conformance to WCAG success criteria. They are for evaluation." sentence that was there before. Was that intentional?

Yes, feedback was that it was confusing in the nav. It is covered in the 2 non rules pages now.

**Comment by shawna-slh** [comment]
I have some suggestions and perspective. I hope to get to it Friday, although I might not until next week.

**Comment by SteveALee** [comment]
> The style on the active page is missing

I don't understand this?

> The empty right column doesn't seem like good use of space. At least some of us preferred the sidebar having the table of contents there.

I'm reversing my opinion on this and think the right nav should always be there - even if it looks a bit messy when there is another box at the head of the page contents. The reason is theCoga Design pattern [Use a Consistent Visual Design ](https://wai-wcag-supplemental.netlify.app/wcag-supplemental/patterns/o1p03-consistent-design/).

@shawna-slh I'l address the layout issue once you present your suggestions.

Stepping back - I do now rather prefer the previous "Zen" clean top header. I wonder if we could provide new viewers the context and about info they require in another way that takes up less vertical space (but not the side menu)

**Comment by shawna-slh** [comment]
Hi @WilcoFiers and ACT TF
Sorry we didn’t provide context before you looked at this. Here’s the situation: This redesign is a small part of a broader project – redesigning Understanding WCAG and Techniques. It also includes the new supplemental guidance.

Understanding WCAG has two levels: guidelines and SCs. We need to provide wayfinding and navigation to both levels. e.g., existing un-designed page: https://www.w3.org/WAI/WCAG22/Understanding/captions-prerecorded
Supplemental Guidance also has two levels under cognitive accessibility (Objectives and Design Patterns). Thus, we have optimized the UI for the two levels. For example: https://wai-wcag-supplemental.netlify.app/wcag-supplemental/patterns/o2p02-site-structure/

Rules was eager to publish before the broader design was finalized, and we did so with an interim design. Now that the broader design is nearly finalized, we want to provide consistency where it makes sense, which means some changes to the interim Rules design.

> The "About Rules" is missing from the navigation

There’s not room for it in the navigation of the others. Given users will probably only read it once then rarely, we tried putting just “About” it in the header. We haven’t tested that with users yet. Hope to soon.
Let’s consider if we want Rules to be consistent with the others with the “About” in the header, or if we want to break consistency since there is room in the navigation area of Rules.

*Optionally*, here are some other things you could put in the navigation bar to be consistent with the main design:
* add <-Previous Rule and Next Rule->
* change “All Rules” that goes to the top of the page, to “All Rules” and “All Proposed Rules” that goes to the specific list on the page -- if you think users might want to jump to the Proposed list directly

(I’m guessing those are not particularly useful for users and thus not something we want to do?)

> The style on the active page is missing

On purpose. Consider that of the 600+ pages using this design (all the techniques, rules, etc.), only a couple are in that nav bar. Thus, changing background of current page could be more confusing than useful. Happy to leave this as an open issue for Phase 2.

> The table of contents on the rules pages is missing the "Test Rules describe ways to test conformance to WCAG success criteria. They are for evaluation." sentence that was there before. Was that intentional?

Yes, per below, it’s important for others that this type of info be in the header.

> We're not clear on the "For developers of evaluation tools and test methodologies" phrase. Is the intent for each page to have the target audience identified in the header? If so we want to discuss this in more details. We briefly talked about "Accessibility testers and test writers", but haven't settled.

It’s important that “not required to meet WCAG” is front and center for techniques and supplemental guidance and understanding. It is now in the header. e.g.:
**Supplemental Guidance to WCAG 2**
**_Additional ways to improve accessibility (not required to meet WCAG)_**

It seems for Rules, the important relevant bit communicates: If you landed here from a search engine, you’re probably not where you want to be. This is for developers of evaluation tools and test methodologies. …

I suggest we open a new issue to work on wording for this.

> Test in the header isn't correctly aligned.

fix in progress.

> The empty right column doesn't seem like good use of space. At least some of us preferred the sidebar having the table of contents there.

fix in progress.

> The top part of the page seems repetitive. This isn't new, but having "Test Rules" repeated 3 times; the header, then the nav, then the h1, that seemed to some of us unnecessarily repetitive.

Note that for Understanding and Supplemental Guidance that wording in the h1 is different (e.g., “Cognitive Accessibility Objective:”, “Cognitive Accessibility Design Pattern:”

A while ago I was going to suggest taking “Test Rule” out of the h1 ; however, I figured out that is where we designated “Proposed Test Rule”. Happy to consider other options for Phase 2.

**Comment by SteveALee** [comment]
@wilco here's the [latest header](https://deploy-preview-86--wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/) and I reverted to always having the side banr.

PS Note that if you use relative URLs in the about and index pages rather than absolute then they will work in the preview as well as the live site. The only downside is if you move pages around the relative links might have to change. But that's unlikely. Maybe we can find a Jekyll code to do it.

**Comment by shawna-slh** [comment]
Hi @WilcoFiers  I think you and Steve exchanged info about this last week. It was my understanding you were going to work on this today/Monday? Hopefully that's just a look around and an, "OK!" :-)

In case Steve didn't make it clear: We would like to publish this ASAP so we can publish the COGA design guide as "Supplemental Guidance". Also, we need confirmation of this design update to apply it to Understanding, Techniques, and ARIA APG.

We can work on the issue of "WCAG 2 Test Rules", maybe ACT Rules, maybe ARIA Test Rules, etc. separately, and implement that when we have agreement from stakeholders.

**Review by shawna-slh — COMMENTED** [review]
Approve after H1 spacing is addressed https://github.com/w3c/wai-wcag-supporting-documents-redesign/issues/94

File: _config.yml [hunk 1]
```diff
@@ -56,15 +56,16 @@ defaults:
...
-        footer:
-          content: Test Rules describe ways to test conformance to WCAG success criteria. They are <a href="../about">for evaluation</a>.
-      standalone_resource_nav_links:
-        - name: All Rules
-          ref: /standards-guidelines/act/rules/
-        - name: About Rules
+      sidebar: true
+      standalone_resource_header:
+        title: WCAG 2 Test Rules
+        subtitle: For developers of evaluation tools and test methodologies
```
Comment (WilcoFiers) [inline _config.yml:62]:
```suggestion
        subtitle: For accessibility testers and developers of test tools and methodologies
```

I think the current wording is too specific. ACT rules are written for anyone doing testing.

File: _layouts/standalone_resource.html [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                {% include_cached icon.html name="different-view" %}
+            </a></li>
+            </ul>
+          </nav>
+          {% endif %}
+        </div>
+    </div>
+    <div class="default-grid with-gap leftcol">
+
+<style>
```
Comment (WilcoFiers) [inline _layouts/standalone_resource.html:97]: Not good practice to put this into inline stylesheets. There's now an act-styles.css file that should be loaded into every page. Better to put it in there.

File: _config.yml [hunk 3]
```diff
@@ -56,15 +56,16 @@ defaults:
...
+        link:
+          title: About WCAG 2 Test Rules
           ref: /standards-guidelines/act/rules/about/
-      standalone_resource_doc_name: "WCAG 2 Test Rules"
+      standalone_resource_pager:
```
Comment (WilcoFiers) [inline _config.yml:66]: Is this the navbar content? Two things about that:

1. Can we add back the link to the "About Rules" page?
2. The styling of the navbar doesn't look right. The active element doesn't have the white background, and when it's not active, this just looks like an inline link, rather than a tab.

File: content/index.md [hunk 4]
```diff
@@ -14,14 +14,13 @@ github:
...
+WCAG 2 Test Rules describe how to test conformance to Web Content Accessibility Guidelines (WCAG) success criteria. They are primarily for developers of evaluation tools and test methodologies. Learn more from [About WCAG 2 Test Rules](/standards-guidelines/act/rules/about/).
 
-This page lists [WCAG 2 Test Rules](https://www.w3.org/WAI/standards-guidelines/act/) formally published by the W3C Accessibility Guidelines Working Group (AGWG). Additional Rules are in development and will be published on this page when completed.
+This page lists Test Rules formally published by the W3C Accessibility Guidelines Working Group (AG WG), and Proposed Test Rules. Additional Rules are in development and will be listed on this page when published.
```
Comment (WilcoFiers) [inline content/index.md:19]: Can we do this editorial work in a separate PR?

File: _layouts/standalone_resource.html [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+        <aside class="box nav-hack sidebar standalone-resource__sidebar ">
+            <header class="box-h ">Page Contents</header>
+            {% capture expanded_content %}{{ content | add_h2_ids }} {% endcapture %}
+            <div class="box-i">
+                <ul>
+                {% assign h2links = expanded_content | get_h2_links %}
+                {% for link in h2links %}
+                <li><a href="#{{ link[1] }}">{{ link[0] }}</a></li>
+                {% endfor %}
+                </ul>
```
Comment (WilcoFiers) [inline _layouts/standalone_resource.html:295]:
```suggestion
                </ul>
                <p>Test Rules describe ways to test conformance to WCAG success criteria. They are <a href="https://www.w3.org/WAI/standards-guidelines/act/rules/about">for evaluation</a>.</p>
```

File: _layouts/standalone_resource.html (hunk 2 above)
Comment (WilcoFiers) [inline _layouts/standalone_resource.html:97]: Let's avoid inline styles. It's not a good practice. There's an act-styles.css file, put it in there.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _config.yml (hunk 3 above)
Comment (SteveALee) [reply _config.yml:66]: The nav bar yaml was refactored. The link to about page is on line 65. We dropped the tabs an iteration back if that's what you mean? 

You can also add paging like the supplemental guidance if you wish...
image

File: _layouts/standalone_resource.html (hunk 2 above)
Comment (SteveALee) [reply _layouts/standalone_resource.html:97]: Why not good practice specifically? Does confuse the cascade for sure. Separate file means separate network load by browser. 

These styles are NOT specific to ACT rule but common to anything that uses the standalone_resource layout. This was how Hidde had it ad I've not reworked to move them to the theme CSS .

File: _config.yml (hunk 1 above)
Comment (SteveALee) [reply _config.yml:62]: @shawna-slh ?

File: _layouts/standalone_resource.html (hunk 2 above)
Comment (shawna-slh) [reply _layouts/standalone_resource.html:97]: > This was how Hidde had it and I've not reworked to move them to the theme CSS.

@WilcoFiers , This seems this is a non-essential change request. I hope you're OK putting it as an issue to consider later. Or, now that you understand it applies more broadly, you are OK leaving it as is.

File: content/index.md (hunk 4 above)
Comment (shawna-slh) [reply content/index.md:19]: Generally, yes. OK for now and any further edits separate?

File: _config.yml (hunk 1 above)
Comment (shawna-slh) [reply _config.yml:62]: Per previous comments, I think this deserves a separate issue. Done: https://github.com/w3c/wcag-act-rules/issues/93

File: _config.yml (hunk 3 above)
Comment (shawna-slh) [reply _config.yml:66]: > Can we add back the link to the "About Rules" page?

It's in the header to be consistent. Additional input welcome in new specific issue https://github.com/w3c/wcag-act-rules/issues/94
